### PR TITLE
Replace `as` conversions with safe ones

### PR DESCRIPTION
**Pull request by petrosagg**
### Motivation

Motivation here https://github.com/MaterializeInc/materialize/pull/16559

This PR removes all `as` conversions from storage crates (not including persist) and some from compute.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

**Review by aljoscha — APPROVED** [review]
seems legit!

**Review by benesch — APPROVED** [review]

File: src/compute-client/src/logging.rs [hunk 1]
```diff
@@ -56,7 +57,7 @@ impl LoggingConfig {
 impl RustType<ProtoLoggingConfig> for LoggingConfig {
     fn into_proto(&self) -> ProtoLoggingConfig {
         ProtoLoggingConfig {
-            interval_ns: Some(self.interval_ns.into_proto()),
+            interval_ns: Some(self.interval.as_nanos().into_proto()),
```
[the remark this answers is not in the document]
Comment (benesch) [reply src/compute-client/src/logging.rs:60]: Doesn't need to block this PR though.

**Review by benesch — APPROVED** [review]
Amazing! Thank you!

File: src/compute-client/src/logging.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (petrosagg) [reply src/compute-client/src/logging.rs:60]: Great idea, fixed!

**Review by benesch — APPROVED** [review]
Oh we already had a ProtoDuration type! Nice!

**Pull request merged**